lib/engine/locations.ex

```diff
@@ -28,51 +30,52 @@ defmodule Engine.Locations do
     end
   end
 
+  @impl true
+  def for_stop(stop_id) do
+    case :ets.lookup(@stop_locations_table, stop_id) do
+      [{^stop_id, locations}] -> locations
+      _ -> []
+    end
+  end
+
```

Comment:
Mainly out of curiosity, did you consider the approach of querying the ETS table by stop_id values within the `vehicle_locations_table`? I haven't played around with ETS all that much, but I think something like that should be doable?